go/vt/vttablet/tabletserver/tx/api.go

```diff
@@ -57,6 +57,7 @@ type (
 		Autocommit      bool
 		Conclusion      string
 		LogToFile       bool
+		Timeout         time.Duration
```

Comment:
how is this timeout different from `StatefulConnection` timeout ?

Reply:
To be honest I'm a bit unclear about this myself. Here's my understanding though:

 * When a stateful connection starts, its timeout may be based on either the `OLTP` or `OLAP` TX timeouts.
 * When a transaction starts, its timeout may be based on either the `OLTP` or `OLAP` TX timeouts, and that timeout may be different than the timeout set on the underlying stateful connection.
 * When the transaction ends, a new `updateTime` is set based on the timeout that was originally set on the stateful connection.

In other words, the timeout associated with the stateful connection persists for the lifetime of that connection, but the timeout associated with a given transaction can change based on the most recent value of `SET workload`.

I'm not really sure if this behavior makes sense. I think the alternative would be, instead of including a `Timeout` with `txProps`, any time a new `Begin()` is called, I could update the `timeout` value on the stateful connection.

Reply:
The one exception to this seems to be reserved connections, which are re-used across TXs. In 6aba785d1a I added code to just update the timeout on existing reserved connections.